h2o-automl/src/main/java/ai/h2o/automl/modeling/StackedEnsembleStepsProvider.java

```diff
@@ -130,6 +138,71 @@ protected boolean isStackedEnsemble(Key<Model> key) {
+                            KeyValue[] mc = (KeyValue[]) model._parms.getClass()
+                                    .getField("_monotone_constraints")
+                                    .get(model._parms);
+                            return mc != null && mc.length > 0;
+                        } catch (NoSuchFieldException | IllegalAccessException e) {
+                            return false;
+                        }
+                    }
+
+                    @Override
+                    protected boolean canRun() {
+                        boolean canRun = super.canRun();
+                        if (!canRun) return false;
+                        int monotoneModels=0;
+                        for (Key<Model> modelKey: getTrainedModelsKeys()) {
+                            if (hasMonotoneConstrains(modelKey))
+                                monotoneModels++;
+                            if (monotoneModels >= 2)
+                                return true;
+                        }
+                        if (monotoneModels == 1) {
+                            aml().job().update(getAllocatedWork().consume(),
+                                    "Only one monotone base model; skipping this StackedEnsemble");
+                            aml().eventLog().info(EventLogEntry.Stage.ModelTraining,
+                                    String.format("Skipping StackedEnsemble '%s' since there is only one monotone model to stack", _id));
+                        } else {
+                            aml().job().update(getAllocatedWork().consume(),
+                                    "No monotone base model; skipping this StackedEnsemble");
+                            aml().eventLog().info(EventLogEntry.Stage.ModelTraining,
+                                    String.format("Skipping StackedEnsemble '%s' since there is no monotone model to stack", _id));
+                        }
+                        return false;
+                    }
+
+                    @Override
+                    @SuppressWarnings("unchecked")
+                    protected Key<Model>[] getBaseModels() {
+                        return Stream.of(getTrainedModelsKeys())
+                                .filter(k -> !isStackedEnsemble(k) && hasMonotoneConstrains(k))
+                                .toArray(Key[]::new);
+                    }
+
+                    @Override
+                    protected StackedEnsembleParameters getStackedEnsembleParameters(Key<Model>[] baseModels, boolean isLast) {
+                        StackedEnsembleParameters params = super.getStackedEnsembleParameters(baseModels, isLast);
+                        GLMModel.GLMParameters metalearnerParams = new GLMModel.GLMParameters();
+                        // Ensure that we have non negative constraint to make it easier to reason about the whole ensemble
+                        // Hard wired in case Metalearner.Algorithm.AUTO changes its implementation
```

Comment:
I understand your concern, but I would still simply use AUTO for now to avoid multiplication of configurations.
We're maintaining both SE and AutoML, so I can't imagine us doing such important changes in SE impl. without checking those used in AutoML.
We should probably keep a comment saying the `non_negative` is a requirement for this particular SE.